doc/COMPONENTS-INTERACTION.md

```diff
@@ -0,0 +1,181 @@
+
+
+## Interaction among Components
+
+
+### Initial loading
+
+* *jangouts* starts initializing and loading other modules from [index.js](../src/app/index.js), which defines the main module *janusHangouts* and its dependency on other modules.
+* It reads the configuration options from *config.json* provided by the server and sets the value of `config` object, provided by [config.provider.js](../src/app/config.provider.js).
+* Then, it directs to `signin` page.
+
+
+### Fetching rooms list
+
+![Fetching Rooms List](fetch-rooms-list.png?raw=true "Fetching Rooms List")
+
+After redirection to signin page, *jh-signin-form.directive.js* calls `RoomService.getRooms()` to get the rooms list. To fetch the list, (which is then displayed on the signin page), the *room.service.js* performs following steps -
+* Janus object is initialized (using `Janus.init()` defined in *janus.js*) after reading the config object provided by *config.provider.js* and a new Janus object is created (using ` new Janus()`). Once created, this object represents a session with the gateway. (See NOTE below this section)
+* Then a new handle on this session is created just to fetch the rooms list.
+* This is done by using `Janus.attach()` and specifying a plugin (`janus.plugin.videoroom` in our case) to recieve a `pluginHandle` object. This allows *jangouts* to exploit the features of a plugin to manipulate the media sent and/or received by a `PeerConnection`
+* We use `pluginHandle` object to send `list` request to Janus server.
+* On success, the `result` object is recieved. The rooms list is accessed as `result.list`, which is an Array of `room` objects.
+* These `room` objects are converted to `Room` objects (*room.factory.js*) for further use.
+* Then `pluginHandle.detach()` destroys the handle.
+
+NOTE: See Janus [JavaScript API](https://janus.conf.meetecho.com/docs/JS.html) for more details on Janus library.
+
+
+### Entering a room
+
+![Enter room](enter-room.png?raw=true "Enter room")
+
+When a participant wants to enter a room, process starts same as fetching rooms list. A new `handle` is created on same `session` (same session used in fetching rooms list). After recieving `pluginHandle` object, *room.service.js* performs following steps -
+ * A new `FeedConnection` object is created with `pluginHandle` object.
+ * Then, it sends a `join` request with `ptype: "publisher"` using `register()` method. On success, it recieves a response object `msg` with `msg.videoroom = "joined"` and a `msg.id` which is unique for each participant in a room.
+ * `ActionService.enterRoom()` is called for creating a new corresponding `Feed` object (with `isPublisher: true`), which is then added to `FeedService.feeds` object.
```

Comment:
Is worth mentioning that this feed gets the role "main". In fact, is the one and only feed with that role.